File: src/Pisaz.Backend.API/Models/Discount/DiscountCode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Pisaz.Backend.API.Models.Discount
{
    public class DiscountCode
    {
        public int Code { get; set; }
        public int Amount { get; set; }
        public int DiscountLimit { get; set; }
        public int UsageCount { get; set; }
        public DateTime ExpirationDate { get; set; }
    }
}

File: src/Pisaz.Frontend/lib/models/discount_code.dart
class DiscountCode {
  final int code;
  final int amount;
  final int? limit;
  final int usageCount;
  final DateTime expirationTime;

  DiscountCode(
      {required this.code,
      required this.amount,
      required this.limit,
      required this.usageCount,
      required this.expirationTime});

  factory DiscountCode.fromJson(Map<String, dynamic> json) {
    return DiscountCode(
        code: json['discountCodeIsGoingToExp'] as int,
        amount: json['amount'] as int,
        limit: json['discountLimit'] as int?,
        usageCount: json['usageCount'] as int,
        expirationTime: DateTime.parse(json['expirationDate']));
  }
}
